framework/src/main/java/org/checkerframework/framework/util/JavaExpressionParseUtil.java

```diff
@@ -119,22 +119,31 @@
     private static final int PARAMETER_REPLACEMENT_LENGTH = PARAMETER_REPLACEMENT.length();
 
     /**
-     * Parse a string and return its representation as a {@link JavaExpression}, or throw a {@link
+     * Parse a string and view-point adapt it to the given {@code context} and {@code localPath}.
```

Comment:
Another possible design, would be to make `localPath` a field of `JavaExpressionContext`.

Reply:
Nit: I would spell it as "viewpoint-adapt" rather than "view-point adapt".  Viewpoint is one word, and viewpoint-adapt is a compound verb (people differ on whether to hyphenate them; I do).